io/src/main/java/io/smallrye/common/io/FileUtils.java

```diff
@@ -0,0 +1,565 @@
+package io.smallrye.common.io;
+
+import static io.smallrye.common.constraint.Assert.checkNotEmptyAfterTrimParam;
+import static io.smallrye.common.constraint.Assert.checkNotNullParam;
+
+import java.io.File;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.AccessMode;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileAttribute;
+
+import io.smallrye.common.constraint.Assert;
+
+/**
+ * Utilities for {@link java.nio.file.Files} using classic
+ * <code>java.io.File</code> as parameter. Different to parameter overload in
+ * PathUtils the return value is File, too. Please prefer a full migration to
+ * java.nio. This is for legacy convenience only.
+ *
+ * @author Boris Unckel
+ *
+ */
+public final class FileUtils {
+
+    /**
+     * A wrapper around {@link java.nio.file.Paths#get(String, String...)} with
+     * explicit null check.
+     *
+     * @param first part of the path, must be not null, not empty and a valid path
+     * @param more optional parts of the path
+     * @return the path constructed out of first and more
+     * @throws IllegalArgumentException if first is null or empty
+     */
+    public static File get(final String first, final String... more) {
+        return Paths.get(checkNotEmptyAfterTrimParam("first", first), more).toFile();
```

Comment:
Thanks for your valuable time to review, I really appreciate that.

- Trimming whitespace at all: On both Windows and Linux/Unixiod systems path elements with leading or trailing space cause trouble. The method does not clean whitespace inside the names. The freedom to use leading/trailing whitespace is still there, but the default tries to avoid it. This is only for Strings the case, all other objects (Path, File, URI) allow it. I have documented the reason with links to detailed documentation in the class javadoc.
- Consistency between "first" and "more" parameter: thanks for the catch, I've fixed it.